shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,240 @@
+  ///             CommitUpdates().
+  ///
+  /// @param[in]  action           A pointer to the custom semantics action
+  /// update.
+  void AddFlutterSemanticsCustomActionUpdate(
+      const FlutterSemanticsCustomAction* action);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Flushes the pending updates and applies them to this
+  ///             accessibility bridge.
+  void CommitUpdates();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the underlying AXTree.
+  AXTree* GetAXTree();
+
+  //------------------------------------------------------------------------------
+  /// @brief      The event generator of this accessibility bridge. It contains
+  ///             the pending accessibility events generated as a result of a
+  ///             semantics update.
+  AXEventGenerator* GetEventGenerator();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the user data.
+  void* GetUserData();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the accessibility bridge delegate.
+  AccessibilityBridgeDelegate* GetDelegate();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the flutter accessibility node with the given id from this
+  ///             accessibility bridge.
+  ///
+  /// @param[in]  id           The id of the flutter accessibility node you want
+  ///                          to retrieve.
+  FlutterAccessibility* GetFlutterAccessibilityFromID(int32_t id) const;
+
+  //------------------------------------------------------------------------------
+  /// @brief      Update the currently focused flutter accessibility node.
+  ///
+  /// @param[in]  id           The id of the currently focused flutter
+  ///                          accessibility node.
+  void SetFocusedNode(int32_t node_id);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the last focused node.
+  int32_t GetLastFocusedNode();
+
+  // AXTreeObserver implementation.
+  void OnNodeWillBeDeleted(AXTree* tree, AXNode* node) override;
+  void OnSubtreeWillBeDeleted(AXTree* tree, AXNode* node) override;
+  void OnNodeCreated(AXTree* tree, AXNode* node) override;
+  void OnNodeDeleted(AXTree* tree, int32_t node_id) override;
+  void OnNodeReparented(AXTree* tree, AXNode* node) override;
+  void OnRoleChanged(AXTree* tree,
+                     AXNode* node,
+                     ax::mojom::Role old_role,
+                     ax::mojom::Role new_role) override;
+  void OnAtomicUpdateFinished(
+      AXTree* tree,
+      bool root_changed,
+      const std::vector<AXTreeObserver::Change>& changes) override;
+
+  std::unique_ptr<AccessibilityBridgeDelegate> delegate_;
```

Comment:
This should definitely be private, and probably be a reference rather than a pointer.